components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -955,67 +976,67 @@ void RRTMGPRadiation::run_impl (const double dt) {
-      auto sw_clrsky_flux_dn_dir_k  = subview_2dk(m_buffer.sw_clrsky_flux_dn_dir_k);
-      auto sw_clnsky_flux_up_k      = subview_2dk(m_buffer.sw_clnsky_flux_up_k);
-      auto sw_clnsky_flux_dn_k      = subview_2dk(m_buffer.sw_clnsky_flux_dn_k);
-      auto sw_clnsky_flux_dn_dir_k  = subview_2dk(m_buffer.sw_clnsky_flux_dn_dir_k);
-      auto lw_clnclrsky_flux_up_k      = subview_2dk(m_buffer.lw_clnclrsky_flux_up_k);
-      auto lw_clnclrsky_flux_dn_k      = subview_2dk(m_buffer.lw_clnclrsky_flux_dn_k);
-      auto lw_clrsky_flux_up_k      = subview_2dk(m_buffer.lw_clrsky_flux_up_k);
-      auto lw_clrsky_flux_dn_k      = subview_2dk(m_buffer.lw_clrsky_flux_dn_k);
-      auto lw_clnsky_flux_up_k      = subview_2dk(m_buffer.lw_clnsky_flux_up_k);
-      auto lw_clnsky_flux_dn_k      = subview_2dk(m_buffer.lw_clnsky_flux_dn_k);
-      auto sw_bnd_flux_up_k  = subview_3dk(m_buffer.sw_bnd_flux_up_k);
-      auto sw_bnd_flux_dn_k  = subview_3dk(m_buffer.sw_bnd_flux_dn_k);
-      auto sw_bnd_flux_dir_k = subview_3dk(m_buffer.sw_bnd_flux_dir_k);
-      auto sw_bnd_flux_dif_k = subview_3dk(m_buffer.sw_bnd_flux_dif_k);
-      auto lw_bnd_flux_up_k  = subview_3dk(m_buffer.lw_bnd_flux_up_k);
-      auto lw_bnd_flux_dn_k  = subview_3dk(m_buffer.lw_bnd_flux_dn_k);
-      auto sfc_flux_dir_vis_k = subview_1dk(m_buffer.sfc_flux_dir_vis_k);
-      auto sfc_flux_dir_nir_k = subview_1dk(m_buffer.sfc_flux_dir_nir_k);
-      auto sfc_flux_dif_vis_k = subview_1dk(m_buffer.sfc_flux_dif_vis_k);
-      auto sfc_flux_dif_nir_k = subview_1dk(m_buffer.sfc_flux_dif_nir_k);
-      auto aero_tau_sw_k     = subview_3dk(m_buffer.aero_tau_sw_k);
-      auto aero_ssa_sw_k     = subview_3dk(m_buffer.aero_ssa_sw_k);
-      auto aero_g_sw_k       = subview_3dk(m_buffer.aero_g_sw_k);
-      auto aero_tau_lw_k     = subview_3dk(m_buffer.aero_tau_lw_k);
+      // Note, ncol will not necessary be m_col_chunk_size because the number of cols
+      // will not always be evenly divided by m_col_chunk_size. In most cases, the
+      // extra space will not cause any problems, but it does sometimes.
+      auto p_lay_k           = subview_2dkc(d_pmid, m_nlay);
+      auto t_lay_k           = subview_2dkc(d_tmid, m_nlay);
+      auto p_lev_k           = subview_2dkc(d_pint, m_nlay+1);
+      auto p_del_k           = subview_2dkc(d_pdel, m_nlay);
+      auto sfc_alb_dir_k     = m_buffer.sfc_alb_dir_k;
+      auto sfc_alb_dif_k     = m_buffer.sfc_alb_dif_k;
+      auto sfc_alb_dir_vis_k = subview_1dkc(d_sfc_alb_dir_vis);
+      auto sfc_alb_dir_nir_k = subview_1dkc(d_sfc_alb_dir_nir);
+      auto sfc_alb_dif_vis_k = subview_1dkc(d_sfc_alb_dif_vis);
+      auto sfc_alb_dif_nir_k = subview_1dkc(d_sfc_alb_dif_nir);
+      auto qc_k              = subview_2dkc(d_qc, m_nlay);
+      auto nc_k              = subview_2dkc(d_nc, m_nlay);
+      auto qi_k              = subview_2dkc(d_qi, m_nlay);
+      auto cldfrac_tot_k     = m_buffer.cldfrac_tot_k;
```

Comment:
cldfrac_tot has special handling if `do_subcol_sampling` is off, so we have to use the buffer.